src/components/EditorsColumn.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import bindAll from 'lodash/bindAll';
+import isEmpty from 'lodash/isEmpty';
+import includes from 'lodash/includes';
+import partial from 'lodash/partial';
+
+import {
+  editorFocusedRequestedLine,
+  updateProjectSource,
+} from '../actions';
+
+import EditorContainer from './EditorContainer';
+import Editor from './Editor';
+
+class EditorsColumn extends React.Component {
+  constructor() {
+    super();
+    bindAll(
+      this,
+      '_handleEditorInput',
+      '_handleRequestedLineFocused',
+    );
+  }
+
+  _allErrorsFor(language) {
+    if (language === 'javascript') {
+      return this.props.errors.javascript.items.
+      concat(this.props.runtimeErrors);
+    }
+
+    return this.props.errors[language].items;
+  }
+
+  _handleEditorInput(language, source) {
+    this.props.actions.updateProjectSource(
+      this.props.currentProject.projectKey,
+      language,
+      source,
+    );
+  }
+
+  _handleRequestedLineFocused() {
+    this.props.actions.editorFocusedRequestedLine();
+  }
+
+  render() {
+    const {currentProject, ui} = this.props;
+    const editors = [];
+    ['html', 'css', 'javascript'].forEach((language) => {
+      if (includes(ui.minimizedComponents, `editor.${language}`)) {
+        return;
+      }
+
+      editors.push(
+        <EditorContainer
+          key={language}
+          language={language}
+          source={currentProject.sources[language]}
+          onMinimize={
+            partial(this.props.onComponentMinimize, `editor.${language}`)
+          }
+        >
+          <Editor
+            errors={this._allErrorsFor(language)}
+            key={language}
+            language={language}
+            percentageOfHeight={1 / editors.length}
+            projectKey={currentProject.projectKey}
+            requestedFocusedLine={ui.editors.requestedFocusedLine}
+            source={currentProject.sources[language]}
+            onInput={partial(this._handleEditorInput, language)}
+            onRequestedLineFocused={this._handleRequestedLineFocused}
+          />
+        </EditorContainer>,
+      );
+    });
+
+    if (isEmpty(editors)) {
+      return null;
+    }
+
+    return (
+      <div className="environment__column">
+        <div className="environment__columnContents editors">{editors}</div>
+      </div>
+    );
+  }
+}
+
+EditorsColumn.propTypes = {
+  actions: React.PropTypes.object.isRequired,
+  currentProject: React.PropTypes.object.isRequired,
+  errors: React.PropTypes.object.isRequired,
+  runtimeErrors: React.PropTypes.array.isRequired,
+  ui: React.PropTypes.object.isRequired,
+  onComponentMinimize: React.PropTypes.func.isRequired,
+};
+
+function mapDispatchToProps(dispatch) {
```

Comment:
Hey, so—using `mapDispatchToProps` generally is a definite :+1: that I’d like to do in `Workspace` as well, BUT the general structure of the components is currently that `Workspace` is the only component that interacts directly with the Redux store, and everything else pushes information up the component tree via callbacks.

I’m definitely open to arguments for an alternative structure in general, but absent a big rethink, I’d like to keep it consistent.

WDYT?